### ISSUE TITLE
[bug]: Collapsible content still accessible when collapsed

### ISSUE THREAD

**Issue by gfellerph** [swisspost/design-system#8296]
### 🐞 Describe the bug

Post collapsible only reduces the height of an element to hide it visually, but does not apply `display: none` or a similar property to hide the content also for keyboard navigation (focusable elements within are still focusable, violating the focus visible criteria) and assistive tech can still access the content in collapsed panels.

The collapsible should actually hide the content upon collapsing in this order:

- State: open
- User toggles
- Play closing animation
- Hide content
- State: collapsed
- User toggles again
- Show content
- Play opening animation
- State: open

### Open questions
- [ ] https://github.com/swisspost/design-system/issues/8297

Hint: if shared functionality is updated, check all instances of usage for regressions
